storybookjs/storybook
# Addon-links: Update linkTo type to accept function

## What I did

Updates the `linkTo()` type to accept a function as the second parameter. This is inline with the [documentation](https://github.com/storybookjs/storybook/tree/next/addons/links#getting-started) and [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/storybook__addon-links/index.d.ts).

## How to test

- Is this testable with Jest or Chromatic screenshots? *No*
- Does this need a new example in the kitchen sink apps? *No*
- Does this need an update to the documentation? *No*

If your answer is yes to any of these, please make sure to include it in your PR.
